api/src/test/java/org/openmrs/api/CohortServiceTest.java

```diff
@@ -644,5 +654,195 @@ public void getMemberships_shouldGetMembershipsAsOfADate() throws Exception {
+		
+		Date dateToTest = dateFormat.parse("2017-01-01 12:00:00");
+		assertTrue(newMember.isMembershipActive(dateToTest));
+		assertTrue(newMember.isMembershipActive(new Date()));
+		assertTrue(newMember.isMembershipActive(startDate));
+	}
+	
+	/**
+	 * @verifies {@link CohortMembership#isMembershipActive(Date)}
+	 */
+	@Test
+	@Verifies(value = "membership should be inactive as of a date", method = "isMembershipActive(Date)")
+	public void isMembershipActive_shouldBeInactiveAsOfDate() throws Exception {
+		CohortMembership newMember = new CohortMembership(new Patient(4));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		newMember.setStartDate(startDate);
+		
+		Date dateToTest = dateFormat.parse("2016-12-01 00:00:00");
+		assertFalse(newMember.isMembershipActive(dateToTest));
+	}
+	
+	/**
+	 * @verifies {@link CohortMembership#isMembershipActive(Date)}
+	 */
+	@Test
+	@Verifies(value = "voided membership should be inactive as of a date", method = "isMembershipActive(Date)")
+	public void isMembershipActive_shouldBeInactiveAsOfDateForVoidedMembership() throws Exception {
+		CohortMembership newMember = new CohortMembership(new Patient(4));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		Date endDate = dateFormat.parse("2017-01-31 00:00:00");
+		newMember.setStartDate(startDate);
+		newMember.setEndDate(endDate);
+		newMember.setVoided(true);
+		
+		Date dateToTest = dateFormat.parse("2017-02-01 00:00:00");
+		assertFalse(newMember.isMembershipActive(dateToTest));
+		Date secondDateToTest = dateFormat.parse("2017-01-15 00:00:00");
+		assertFalse(newMember.isMembershipActive(secondDateToTest));
+	}
+	
+	/**
+	 * @verifies {@link CohortMembership#compareTo(CohortMembership)}}
+	 */
+	@Test
+	@Verifies(value = "should compare two membership objects", method = "compareTo(CohortMembership)")
+	public void compareTo_shouldCompareMemberships() throws Exception {
+		CohortMembership aMember = new CohortMembership(new Patient(4));
+		CohortMembership bMember = new CohortMembership(new Patient(4));
+		assertEquals(aMember.compareTo(bMember), 0);
+		assertEquals(bMember.compareTo(aMember), 0);
+	}
+	
+	/**
+	 * @see CohortService#getCohortsContainingPatient(Patient)
+	 *
+	 */
+	@Test
+	@Verifies(value = "should not return voided memberships", method = "getCohortsContainingPatient(Patient)")
+	public void getCohortsContainingPatient_shouldNotReturnVoidedMemberships() throws Exception {
+		executeDataSet(COHORT_XML);
+		
+		Cohort cohort = service.getCohort(2);
+		
+		// add a patient to cohorts
+		Patient patient = new Patient(7);
+		CohortMembership membership = new CohortMembership(patient);
+		cohort.addMembership(membership);
+		assertTrue(cohort.contains(patient.getPatientId()));
+		
+		// void the membership
+		membership.setVoided(true);
```

Comment:
I agree, typically you add comments to areas where you're implementing some logic that isn't straight forward to understand, doesn't make sense to comment on method call like this unless you're saying why you are calling it.